### PR TITLE
New jmap tool

### PR DESCRIPTION
**Pull request by SueChaplain**
Topic to describe the use of the jmap tool, which is added
for compatibility with the reference implmentation.

Closes: #301

Signed-off-by: Sue Chaplain <sue_chaplain@uk.ibm.com>

### CODE REVIEW

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by SueChaplain** [comment]
@pdbain-ibm @pshipton 

I downloaded a nightly docker to try out jmap and answer questions like "how do you pass multiple IDs to the program". I noticed that the help for the tool is slightly different from the input in the issue, and aligns with the jps topic (references VMID instead of PS) so I made changes accordingly. I was also able to get some sample output, although I leave it to you as to whether this is suitable or needs updating.

Content is staged: https://eclipse.github.io/openj9-docs-staging/313/version0.15/#new-java-memory-map-tool

File: docs/tool_jmap.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  6            342           8208    java.lang.StringBuilder
+  7            151           7248    jdk.internal.org.objectweb.asm.Item
+  8            396           6336    [Ljava.lang.Object;
+```
+
+<i class="fa fa-exclamation-triangle" aria-hidden="true"></i> **Restrictions:** This tool is not supported and is subject to change or removal in future releases. Although similar in usage and output to the HotSpot tool of the same name, this tool is a different implementation that is specific to OpenJ9.
+
+The following tool limitations apply:
+
+- Displays information only for local processes that are owned by the current user, due to security considerations. You can display information for remote processes by  using `ssh jmap <options> <pid>`.
```
Comment (DanHeidinga) [inline docs/tool_jmap.md]:
```suggestion
- Displays information only for local processes that are owned by the current user, due to security considerations. You can display information for remote processes by  using `ssh user@host jmap <options> <pid>`.
```

File: docs/tool_jmap.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+* [1] https://www.gnu.org/software/classpath/license.html
+* [2] http://openjdk.java.net/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Java memory map (`jmap`) tool
+
+Use the `jmap` tool to get memory information for a particular Java&trade; process, or list of processes. The tool shows information for every Java process that is owned by the current user ID on the current host. The command syntax is as follows:
```
Comment (pshipton) [inline docs/tool_jmap.md]: Not sure about `The tool shows information for every Java process that is owned by the current user ID on the current host.`. Should this say the tool *can* show information...? This seems like it was copied from jps, but it doesn't apply to jmap.

Comment (SueChaplain) [reply docs/tool_jmap.md]: Hmm ... yes. This was started by someone else last week and I took over yesterday - assumed what was there so far was received as input, although I did strip out some things that didn't seem to apply. Will change to

>The tool shows statistics about classes on the heap, including the number of objects and their aggregate size.

File: docs/tool_jmap.md (hunk 1 above)
Comment (SueChaplain) [reply docs/tool_jmap.md]: Added

**Pull request merged**